sw/device/tests/edn_auto_mode.c

```diff
@@ -0,0 +1,141 @@
+#define OTBN_RANDOMNESS_ITERATIONS 10
+#define TEST_PROCEDURE_REPETITIONS 2
+
+static dif_entropy_src_t entropy_src;
+static dif_csrng_t csrng;
+static dif_edn_t edn0;
+static dif_edn_t edn1;
+static dif_aes_t aes;
+static dif_otbn_t otbn;
+static dif_rv_core_ibex_t rv_core_ibex;
+
+// AES ECB encryption transaction.
+static dif_aes_transaction_t transaction = {
+    .operation = kDifAesOperationEncrypt,
+    .mode = kDifAesModeEcb,
+    .key_len = kDifAesKey256,
+    .key_provider = kDifAesKeySoftwareProvided,
+    .mask_reseeding = kDifAesReseedPerBlock,
+    .manual_operation = kDifAesManualOperationAuto,
+    .reseed_on_key_change = true,
+    .ctrl_aux_lock = false,
+};
+
+OTTF_DEFINE_TEST_CONFIG();
+
+// Initializes the peripherals used in this test.
+static void init_peripherals(void) {
+  CHECK_DIF_OK(dif_entropy_src_init(
+      mmio_region_from_addr(TOP_EARLGREY_ENTROPY_SRC_BASE_ADDR), &entropy_src));
+  CHECK_DIF_OK(dif_csrng_init(
+      mmio_region_from_addr(TOP_EARLGREY_CSRNG_BASE_ADDR), &csrng));
+  CHECK_DIF_OK(
+      dif_edn_init(mmio_region_from_addr(TOP_EARLGREY_EDN0_BASE_ADDR), &edn0));
+  CHECK_DIF_OK(
+      dif_edn_init(mmio_region_from_addr(TOP_EARLGREY_EDN1_BASE_ADDR), &edn1));
+  CHECK_DIF_OK(
+      dif_aes_init(mmio_region_from_addr(TOP_EARLGREY_AES_BASE_ADDR), &aes));
+  CHECK_DIF_OK(
+      dif_otbn_init(mmio_region_from_addr(TOP_EARLGREY_OTBN_BASE_ADDR), &otbn));
+  CHECK_DIF_OK(dif_rv_core_ibex_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR),
+      &rv_core_ibex));
+}
+
+static void configure_otbn(void) {
+  otbn_randomness_test_start(&otbn, OTBN_RANDOMNESS_ITERATIONS);
+}
+
+// configure the entropy complex
+static void entropy_config(void) {
+  dif_edn_auto_params_t edn_params0 = edn_testutils_auto_params_build(true);
+  dif_edn_auto_params_t edn_params1 = edn_testutils_auto_params_build(true);
+  // Disable the entropy complex
+  CHECK_STATUS_OK(entropy_testutils_stop_all());
+  // Enable ENTROPY_SRC in FIPS mode
+  CHECK_DIF_OK(dif_entropy_src_configure(
+      &entropy_src, entropy_testutils_config_default(), kDifToggleEnabled));
+  // Enable CSRNG
+  CHECK_DIF_OK(dif_csrng_configure(&csrng));
+  // Enable EDNs in auto request mode
+  CHECK_DIF_OK(dif_edn_set_auto_mode(&edn0, edn_params0));
+  CHECK_DIF_OK(dif_edn_set_auto_mode(&edn1, edn_params1));
+}
+
+static status_t stress_test_edns(void) {
+  int otbn_execute_rounds = OTBN_ROUNDS;
+  dif_rv_core_ibex_rnd_status_t ibex_rnd_status;
+  dif_otbn_status_t otbn_status;
+  uint32_t ibex_rnd_data;
+  while (otbn_execute_rounds) {
+    CHECK_DIF_OK(dif_otbn_get_status(&otbn, &otbn_status));
+    if (otbn_status == kDifOtbnStatusIdle) {
+      configure_otbn();
+      otbn_execute_rounds--;
+    }
+    if (aes_testutils_get_status(&aes, kDifAesStatusOutputValid))
+      CHECK_STATUS_OK(aes_testutils_setup_encryption(transaction, &aes));
+
+    CHECK_DIF_OK(
+        dif_rv_core_ibex_get_rnd_status(&rv_core_ibex, &ibex_rnd_status));
```

Comment:
The build in CI was not able to find this symbol.

Reply:
Thanks, I'll have a look.